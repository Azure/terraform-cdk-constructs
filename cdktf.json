{
  "language": "typescript",
  "app": "npx ts-node ./src/examples/AzureResourceGroup.ts",
  "terraformProviders": ["Azure/azapi@~> 0.6.0"],
  "projectId": "1b5f16bd-67ac-4af6-a33f-7bcafff141eb",
  "context": {
    "excludeStackIdFromLogicalIds": "true",
    "allowSepCharsInLogicalIds": "true"
  }
}