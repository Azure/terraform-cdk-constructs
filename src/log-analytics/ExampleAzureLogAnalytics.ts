import { AzureLogAnalytics } from '.';
import { App, TerraformStack} from "cdktf";
import {ResourceGroup} from "@cdktf/provider-azurerm/lib/resource-group";
import {AzurermProvider} from "@cdktf/provider-azurerm/lib/provider";
import { Construct } from 'constructs';



const app = new App();
export class exampleAzureLogAnalytics extends TerraformStack {
  constructor(scope: Construct, id: string) {
    super(scope, id);
    
    new AzurermProvider(this, "azureFeature", {
        features: {},
      });

    const rg = new ResourceGroup(this, "rg", {
      location: 'eastus',
      name: 'rg-latest',

    });

    new AzureLogAnalytics(this, 'testLA', {
      name: 'la-test',
      location: 'eastus',
      retention: 90,
      sku: "PerGB2018",
      resource_group_name: rg.name ,
    });
  }
}

new exampleAzureLogAnalytics(app, "testAzureLogAnalytics");

app.synth();