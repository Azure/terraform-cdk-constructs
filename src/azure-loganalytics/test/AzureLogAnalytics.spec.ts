import { Testing, TerraformStack} from 'cdktf';
import { exampleAzureLogAnalytics} from './ExampleAzureLogAnalytics'
import 'cdktf/lib/testing/adapters/jest';


describe('AzureLogAnalytics-Snapshot', () => {
  it('renders a AzureLogAnalytics', () => {
    const synthed = Testing.synthScope((stack) => {
      new exampleAzureLogAnalytics(stack, "testAzureLogAnalytics");
    });
    expect(synthed).toMatchSnapshot();
  });
});


describe("AzureLogAnalytics-Terraform", () => {
  it("check if the produced terraform configuration is valid", () => {
    const app = Testing.app();
    const stack = new TerraformStack(app, "test");
    
    Testing.synthScope((stack) => {
      new exampleAzureLogAnalytics(stack, "testAzureLogAnalytics");
    });

    // We need to do a full synth to validate the terraform configuration
    expect(Testing.fullSynth(stack)).toBeValidTerraform();
  });

  it("check if this can be planned", () => {
    const app = Testing.app();
    const stack = new TerraformStack(app, "test");

    Testing.synthScope((stack) => {
      new exampleAzureLogAnalytics(stack, "testAzureLogAnalytics");
    });

    // We need to do a full synth to plan the terraform configuration
    expect(Testing.fullSynth(stack)).toPlanSuccessfully();
  });
});
