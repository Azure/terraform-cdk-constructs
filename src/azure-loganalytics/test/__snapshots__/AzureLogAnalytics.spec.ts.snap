// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AzureLogAnalytics-Snapshot renders a AzureLogAnalytics 1`] = `
"{
}"
`;
